### PR TITLE
[WIP] feat(core): add -noPolyfill option to `grunt build`

### PR DESCRIPTION
**Pull request by straker**
Reduces the file size of axe from 893KB to 731KB unminified, and 450KB to 384KB minified.

Partially closes issue: #1935 

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

### CODE REVIEW

**Comment by WilcoFiers** [comment]
Having the build option doesn't fully resolve the issue. We should have these no-poly files published to NPM (and included in sri-history.json)

File: build/polyfills-generator.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+const outputFile = path.join(
+	__dirname,
+	'..',
+	'tmp',
+	'core',
+	'polyfills',
+	'index.js'
+);
+
+async function run() {
```
Comment (jkodu) [inline build/polyfills-generator.js]: This is a clone of `imports-generator` - https://github.com/dequelabs/axe-core/blob/develop/build/imports-generator.js

I see no reason it has to be though. You should make this a common function, that when given a source and destination and the namespace to mount, it creates these files.

Why have two files with the same code one for creating `axe.imports` and one for creating `axe.polyfills`?

File: lib/core/polyfills/index.js [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+require('weakmap-polyfill');
+
+/* eslint-disable */
+/*
+ These pollyfills came directly from the ES Specification it's self
+ Contained within:
+  - Object.assign
+  - Array.prototype.find
+*/
+if (typeof Object.assign !== 'function') {
```
Comment (jkodu) [inline lib/core/polyfills/index.js]: another thought is, given we have `core-js` already above for `reduce`, can we remove the copy pasted polyfills below and swap them out to features from `core-js`

Eg: `core-js(-pure)/es|stable|features/object/assign`

Comment (stephenmathieson) [reply lib/core/polyfills/index.js]: I agree here, but don't think that needs to change for this PR to land

**Pull request closed without merging**